Source/EventFlow.EntityFramework/Extensions/DbContextExtensions.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+
+namespace EventFlow.EntityFramework.Extensions
+{
+    public static class Bulk
```

Comment:
I'm not proud of this one. Please help. See comment for further information.

Reply:
There is no way to delete entities without querying them first: https://www.google.de/search?q=ef+core+"delete+where"

There seems to be a package that supports batch delete, but 1.) that's lots of code, 2.) it needs a reference to `Microsoft.EntityFrameworkCore.Relational` (that we don't need otherwise) - and 3.) obviously it needs to hardcode SQL commands for the major providers. Which means it only works for those, probably: https://github.com/zzzprojects/EntityFramework-Plus/blob/master/src/shared/Z.EF.Plus.BatchDelete.Shared/BatchDelete.cs

This affects both Read Stores and Snapshot Stores.

Reply:
Sorry, no help here, haven't worked with EF in almost five years. Pretty sure I'm outdated (we use Dapper for almost everything)